sklearn/semi_supervised/_self_training.py

```diff
@@ -291,107 +371,185 @@ def fit(self, X, y):
+                See :ref:`Metadata Routing User Guide <metadata_routing>` for
+                more details.
+
         Returns
         -------
         y : ndarray of shape (n_samples,)
             Array with predicted labels.
         """
         check_is_fitted(self)
+        _raise_for_params(params, self, "predict")
+
+        if _routing_enabled():
+            # metadata routing is enabled.
+            routed_params = process_routing(self, "predict", **params)
+        else:
+            routed_params = Bunch(estimator=Bunch(predict={}))
+
         X = self._validate_data(
             X,
             accept_sparse=True,
             force_all_finite=False,
             reset=False,
         )
-        return self.base_estimator_.predict(X)
+        return self.estimator_.predict(X, **routed_params.estimator.predict)
 
     @available_if(_estimator_has("predict_proba"))
-    def predict_proba(self, X):
+    def predict_proba(self, X, **params):
         """Predict probability for each possible outcome.
 
         Parameters
         ----------
         X : {array-like, sparse matrix} of shape (n_samples, n_features)
             Array representing the data.
 
+        **params : dict of str -> object
+            Parameters to pass to the underlying estimator's predict_proba method.
+
+            .. versionadded:: 1.6
+                Only available if `enable_metadata_routing=True`,
+                which can be set by using
+                ``sklearn.set_config(enable_metadata_routing=True)``.
+                See :ref:`Metadata Routing User Guide <metadata_routing>` for
+                more details.
+
         Returns
         -------
         y : ndarray of shape (n_samples, n_features)
             Array with prediction probabilities.
         """
         check_is_fitted(self)
+        _raise_for_params(params, self, "predict_proba")
+
+        if _routing_enabled():
+            # metadata routing is enabled.
+            routed_params = process_routing(self, "predict_proba", **params)
+        else:
+            routed_params = Bunch(estimator=Bunch(predict_proba={}))
+
         X = self._validate_data(
             X,
             accept_sparse=True,
             force_all_finite=False,
             reset=False,
         )
-        return self.base_estimator_.predict_proba(X)
+        return self.estimator_.predict_proba(X, **routed_params.estimator.predict_proba)
 
     @available_if(_estimator_has("decision_function"))
-    def decision_function(self, X):
-        """Call decision function of the `base_estimator`.
+    def decision_function(self, X, **params):
+        """Call decision function of the `estimator`.
 
         Parameters
         ----------
         X : {array-like, sparse matrix} of shape (n_samples, n_features)
             Array representing the data.
 
+        **params : dict of str -> object
+            Parameters to pass to the underlying estimator's decision_function method.
```

Comment:
```suggestion
            Parameters to pass to the underlying estimator's ``decision_function`` method.
```

and similar for the other ones.

Reply:
[feff521](https://github.com/scikit-learn/scikit-learn/pull/28494/commits/feff5219abc89dfc80b88bcbe3bff51e77e450f7)